test/fw/ptl/utils/pbs_testsuite.py

```diff
@@ -828,6 +829,327 @@ def init_proc_mon(self):
+
+        :param conf: the pbs.conf variable
+        :type conf: str
+        :param primary_host: hostname of the primary server host
+        :type primary_host: str
+        :param hosttype: type of host being reverted
+        :type hosttype: str
+        :param hostobj: PTL object associated with the host
+        :type hostobj: PBSService
+
+        :return default value of the pbs.conf variable if it can be determined
+        as a string, otherwise None
+        """
+        if conf == "PBS_SERVER":
+            return primary_host
+        elif conf == "PBS_START_SCHED":
+            if hosttype == "server":
+                return "1"
+            else:
+                return "0"
+        elif conf == "PBS_START_COMM":
+            if hosttype == "comm":
+                return "1"
+            else:
+                return "0"
+        elif conf == "PBS_START_SERVER":
+            if hosttype == "server":
+                return "1"
+            else:
+                return "0"
+        elif conf == "PBS_START_MOM":
+            if hosttype == "mom":
+                return "1"
+            else:
+                return "0"
+        elif conf == "PBS_CORE_LIMIT":
+            return "unlimited"
+        elif conf == "PBS_SCP":
+            scppath = self.du.which(hostobj.hostname, "scp")
+            if scppath != "scp":
+                return scppath
+
+        return None
+
+    def _revert_pbsconf_comm(self, primary_server, vals_to_set):
+        """
+        Helper function to revert_pbsconf to revert all comm daemons' pbs.conf
+
+        :param primary_server: object of the primary PBS server
+        :type primary_server: PBSService
+        :param vals_to_set: dict of pbs.conf values to set
+        :type dict
+        """
+        for comm in self.comms.values():
+            if comm.hostname == primary_server.hostname:
+                continue
+
+            new_pbsconf = dict(vals_to_set)
+            restart_comm = False
+            pbs_conf_val = self.du.parse_pbs_config(comm.hostname)
+            if not pbs_conf_val:
+                raise ValueError("Could not parse pbs.conf on host %s" %
+                                 (comm.hostname))
+
+            # to start with, set all keys in new_pbsconf with values from the
+            # existing pbs.conf
+            keys_to_delete = []
+            for conf in new_pbsconf:
+                if conf in pbs_conf_val:
+                    new_pbsconf[conf] = pbs_conf_val[conf]
+                else:
+                    # existing pbs.conf doesn't have a default variable set
+                    # Try to determine the default
+                    val = self._get_dflt_pbsconfval(conf,
+                                                    primary_server.hostname,
+                                                    "comm", comm)
+                    if val is None:
+                        self.logger.error("Couldn't set %s in pbs.conf" %
```

Comment:
> Using couldn't feels like that the program tried but it couldn't make it

Well, that actually is what's happening though, we try to find the default value for this variable in _get_dflt_pbsconfval(), but if that returns None, we say that we tried but couldn't figure out how to do it. Does it make sense now?

Reply:
Well, I'm ok if you want to retain it the way it is now. I thought, in this case, the log could be that it couldn't figure out a default value for the configuration. 

To me, the current log message just feels like a thing to say when setup tried putting the configuration in the conf file but couldn't do so because of factors like unable to open the file or file not present or write failed etc.

Reply:
ok, i see what you are saying, alright I'll change it to say "Couldn't revert %s in pbs.conf to its default value"